pkg/logutil/logutil.go

```diff
@@ -28,3 +30,54 @@ func WithPluginConstants(
 		"configDigest", configDigest,
 	)
 }
+
+type ContextKey string
+
+const (
+	// SeqNrKey refers to the OCR sequence number.
+	// Linter complains if this is a plain "string" type.
+	SeqNrKey ContextKey = "ocrSeqNr"
+
+	SeqNrLoggerKey = "ocrSeqNr"
+)
+
+// CtxAndLoggerWithSeqNr returns a context.Context and logger with the sequence number set.
+func CtxAndLoggerWithSeqNr(
+	ctx context.Context,
+	lggr logger.Logger,
+	seqNr uint64,
+) (context.Context, logger.Logger) {
+	newCtx := CtxWithSeqNr(ctx, seqNr)
+	newLggr := LoggerWithSeqNr(newCtx, lggr)
+	return newCtx, newLggr
+}
+
+// CtxWithSeqNr returns a context.Context with the sequence number set.
+func CtxWithSeqNr(ctx context.Context, seqNr uint64) context.Context {
+	return context.WithValue(ctx, SeqNrKey, seqNr)
+}
+
+// LoggerWithSeqNr returns a logger with the sequence number set to the correct log field key.
+// It assumes that the given ctx has the sequence number set, if not the field will be set to
+// 0 in the log field.
+func LoggerWithSeqNr(
+	ctx context.Context,
+	lggr logger.Logger,
+	keyvals ...any,
+) logger.Logger {
+	return logger.With(
+		lggr,
+		append(keyvals, SeqNrLoggerKey, GetSeqNr(ctx))...,
```

Comment:
Is it useful to have the seqNr field if `GetSeqNr` returns zero?

Reply:
Initially I'd say yeah because it'll let us know where it isn't being correctly piped.